tests/integration/protected_images.t

```diff
@@ -0,0 +1,103 @@
+#!/usr/bin/perl -w
+
+use ProductOpener::PerlStandards;
+
+use Test::More;
+use ProductOpener::APITest qw/:all/;
+use ProductOpener::Products qw/:all/;
+use ProductOpener::Test qw/:all/;
+use ProductOpener::TestDefaults qw/:all/;
+
+use File::Basename "dirname";
+
+use Storable qw(dclone);
+
+remove_all_users();
+
+remove_all_products();
+
+wait_application_ready();
+my $sample_products_images_path = dirname(__FILE__) . "/inputs/upload_images";
+
+# Create an admin
+my $admin_ua = new_client();
+my $resp = create_user($admin_ua, \%admin_user_form);
+ok(!html_displays_error($resp));
+
+# Create a normal user
+my $ua = new_client();
+my %create_user_args = (%default_user_form, (email => 'bob@gmail.com'));
+$resp = create_user($ua, \%create_user_args);
+ok(!html_displays_error($resp));
+
+my %product_form = (
+	generic_name => "A generic name",
+	ingredients_text => "apple, milk, eggs, palm oil",
+	categories => "some unknown category",
+	labels => "organic",
+	origin => "france",
+	brands => "Test brand",
+	packaging_text_en =>
+		"1 wooden box to recycle, 6 25cl glass bottles to reuse, 3 steel lids to recycle, 1 plastic film to discard",
+	'nutriment_energy-kj' => 10,
+	nutriment_fat => 5,
+	nutriment_proteins => 2,
+	quantity => "90g",
+);
+
+my @products = (
+	{
+		%{dclone(\%default_product_form)},
+		%{dclone(\%product_form)},
+		(
+			code => '0900000000139',
+			product_name => "A test product",
+		)
+	},
+);
+# create the products in the database
+foreach my $product_form_override (@products) {
+	edit_product($ua, $product_form_override);
+}
+
+# Setting the owner of the product
+my $product_ref = retrieve_product('0900000000139');
+$product_ref->{owner} = "sample", store_product("sample", $product_ref, "protecting image");
```

Comment:
It's a good thing to set the owner to "sample" that way. Then for the following tests, you need 2 users: one user "sample" which is the owner, who should upload an image: the image should be protected as it's uploaded and selected by the owner.

And then you need to use another user (like the "tests" user) to try to upload and select a different image: it should not work as the image is protected.